src/Microsoft.OpenApi/Validations/ValidationRule.cs

```diff
@@ -37,11 +42,12 @@ public abstract class ValidationRule
     public class ValidationRule<T> : ValidationRule where T : IOpenApiElement
     {
         private readonly Action<IValidationContext, T> _validate;
-
+        
         /// <summary>
         /// Initializes a new instance of the <see cref="ValidationRule"/> class.
         /// </summary>        
         /// <param name="validate">Action to perform the validation.</param>
+        [Obsolete("Please use the other constructor and specify a name")]
         public ValidationRule(Action<IValidationContext, T> validate)
             : this (null, validate)
```

Comment:
```suggestion
            : this (Guid.NewGuid().ToString("D"), validate)
```

This way we avoid any side-effect during the remove call.

Reply:
will do that